src/lib/validate-input.js

```diff
@@ -653,12 +653,26 @@ export function validateScatterElementsParams(input, indices, updates, {axis = 0
         `The axis ${axis} should be an unsigned integer in the interval [0, ${inputRank}).`);
   }
   const axisSize = input.shape[axis];
+  const updatedLocationDict = {};
   for (let i = 0; i < sizeOfShape(indices.shape); ++i) {
-    const index = indices.getValueByIndex(i);
+    let index = indices.getValueByIndex(i);
     if (!Number.isInteger(index) || index < -axisSize || index >= axisSize) {
       throw new Error(`Invalid indices value - it should be an integer in the interval ` +
       `[${-axisSize}, ${axisSize - 1}]`);
     }
+    const indicesLocation = indices.locationFromIndex(i);
+    const originOutputLocation =
+        [...indicesLocation.slice(0, axis), index, ...indicesLocation.slice(axis + 1)];
+    index = index < 0 ? index + input.shape[axis] : index;
+    const outputLocation =
+        [...indicesLocation.slice(0, axis), index, ...indicesLocation.slice(axis + 1)];
+    const locationString = outputLocation.toString();
+    if (Object.hasOwn(updatedLocationDict, locationString)) {
```

Comment:
Does a [set of locations](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) work here? It would be conceptually clearer than Object.hasOwn, and it's less to store (there's no key:value mapping, just boolean presence/absence).

Reply:
> Does a [set of locations](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) work here? It would be conceptually clearer than Object.hasOwn, and it's less to store (there's no key:value mapping, just boolean presence/absence).

Thanks @fdwr! Yes, the **set** works for `if` condition.
Here I use \<key : value> to also record the location information when throwing the error message with current checking location and previous ones like codes on Line671-672 . PTAL, thanks!

Reply:
Aha, so the error message includes the *previous* location - got it.

Reply:
I prefer to using `Set` as well. Error message could just contain the problematic location, i.e. ` Invalid indices, [${originOutputLocation}] is not unique.`

Reply:
Updated in new commits. Please take another look, thanks!